apis/sql/v1beta1/sqlinstance_types.go

```diff
@@ -441,7 +441,38 @@ type InstanceValueFrom struct {
 	SecretKeyRef *v1alpha1.SecretKeyRef `json:"secretKeyRef,omitempty"`
 }
 
+type BinLogCoordinates struct {
+	/* Name of the binary log file for a Cloud SQL instance. */
+	BinLogFileName string `json:"binLogFileName,omitempty"`
+
+	/* Position (offset) within the binary log file. */
+	BinLogPosition int64 `json:"binLogPosition,omitempty,string"`
+}
+
+type CloneSource struct {
+	/* Binary log coordinates, if specified, identify the position up to which the source instance is
+	cloned. If not specified, the source instance is cloned up to the most recent binary log coordinates. */
+	// +optional
+	BinLogCoordinates *BinLogCoordinates `json:"binLogCoordinates,omitempty"`
+
+	/* (SQL Server only) Clone only the specified databases from the source instance. Clone all databases if empty. */
+	// +optional
+	SQLDatabaseRefs []refsv1beta1.SQLDatabaseRef `json:"sqlDatabaseRefs,omitempty"`
+
+	/* Timestamp, if specified, identifies the time to which the source instance is cloned. */
+	// +optional
+	PointInTime *string `json:"pointInTime,omitempty"`
+
+	/* The source SQLInstance to clone */
+	SQLInstanceRef refsv1beta1.SQLInstanceRef `json:"sqlInstanceRef,omitempty"`
+}
+
 type SQLInstanceSpec struct {
+	/* Create this database as a clone of a source instance. Immutable. */
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec.cloneSource is immutable"
```

Comment:
A silly question: Does the type of the field matter for the validation? E.g. is it able to understand "equals" for object type here?

Reply:
Yes, equality works for maps: https://github.com/google/cel-spec/blob/master/doc/langdef.md#equality. I also tested this manually.

Reply:
I am a little wary of introducing CEL (1) on a high profile resource and (2) accidentally i.e. as part of a wider PR.  (2) is in case we need to revert and so that we can focus on CEL.

Reply:
I proposed using CEL for immutable check in the [CR Validation doc](https://docs.google.com/document/d/1nKB6hkMV34K0SVtOdFyGZ48D4NhKhGAOTGXPEVLNAqQ/edit?resourcekey=0-hbpIoeozx9it1GVTZDIbBw&tab=t.0#bookmark=id.6quka07soumj) and put the reasons on how the decision is made:

- Kubernetes supports CEL since 1.26 which is the oldest GKE cluster version. Using CEL shall not cause GKE cluster version issues, but we shall reevaluate this when the condition changes.
- Some KCC resources use webhook to validate immutable fields. We can continue using that, but for external contributors CEL is much easier for self-learning. 
- We only apply CEL on immutable fields. More complicated CEL validation requires further discussion.

Regarding to Justin's concerns, +1 to split the immutable check in a separate PR. Because 1. we do want to use CEL for immutable field check and makes it as the trends. 2. users may hit the CEL-supportability problem when they install KCC in a non-GKE cluster and the k8s is very old. Though the possibility of this case itself is low, SQL is a high profile resource like Justin mentioned. So splitting the PR is a good idea, we can still use CEL and keep us with the clean rollback options.

Reply:
Ok, moving this CEL validation to a separate PR.